lib/utils/job-utils/racadm-parser.js

```diff
@@ -0,0 +1,229 @@
+// Copyright 2016, EMC, Inc.
+
+"use strict";
+
+var di = require('di');
+
+module.exports = parseRacadmDataFactory;
+di.annotate(parseRacadmDataFactory, new di.Provide('JobUtils.RacadmCommandParser'));
+di.annotate(parseRacadmDataFactory,
+    new di.Inject(
+        'Assert',
+        '_'
+    )
+);
+
+function parseRacadmDataFactory(
+    assert,
+    _
+) {
+
+    function RacadmCommandParser() {}
+
+    /**
+     * Parser software inventory list to get standard software list json format.
+     *
+     * @param {string} softwareListData
+     */
+    RacadmCommandParser.prototype.getSoftwareList = function(softwareListData) {
+        var softwareInventory = {};
+
+        var lines = softwareListData.trim().split('\n');
+        var filteredLines = _.filter(lines, function(line){
+            return line.indexOf('=') !== -1 && line.indexOf('ComponentType') === -1;
+        });
+
+        if (filteredLines.length % 4 !== 0){
+            throw "software list data is not aligned in correct way";
```

Comment:
it's not a good practice to throw a string instead of an Error instance, see the discussion in stackoverflow: http://stackoverflow.com/questions/11502052/throwing-strings-instead-of-errors